### ISSUE TITLE
Export data not working on iPad

### ISSUE THREAD

**Issue by sychang** [Recidiviz/pulse-dashboard#622]
**Describe the bug**
A PA user was unable to export data from any of the charts in the Lantern dashboard on iPad. They tried downloading each chart but saw nothing in their downloads folder. Not sure if this is WAI, given iPad. If it is WAI, we could consider adding documentation to 

This was the followup note from the PA user: "Good morning Serena. I was again unsuccessful in exporting from the dashboard while using Google Chrome version 86.0.4240.93 on an  I checked my download folders as advised and saw nothing." 

**To Reproduce**
1. On iPad, go to any chart on the Lantern PA dashboard
2. Click on the three dots and "export data"
3. Nothing happens

**Expected behavior**
Data should appear in a downloads destination on the ipad.

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6]: iPad Pro (3rd generation)
 - OS: [e.g. iOS8.1]: unknown
 - Browser [e.g. stock browser, safari]: Chrome
 - Version [e.g. 22]: 86.0.4240.93

**Additional context**
Not sure if this is WAI, given iPad.

**Comment by jovergaag** [comment]
This is not reproducible in Chrome Dev Tools. @Wmaileq let me know if you do not have access to an iPad emulator.

**Comment by Wmaileq** [comment]
Reproduced in IOS Chrome.

**Comment by Wmaileq** [comment]
Investigations were made here. It looks like IOS webkit could not download files due to limitations of the system. (mentioned here: https://github.com/eligrey/FileSaver.js/#ios) 
The one workaround is to temporarily upload created zip\csv to the server and immediately download.

**Comment by jovergaag** [comment]
@AlexK-ino This is the next task to get started on. @Wmaileq was able to reproduce it so it you have trouble reproducing he might be able to help. 

Thank you!

**Comment by AlexK-ino** [comment]
Hello, @jovergaag!
Yes, I have reproduced this bug, except it is not only on iPad but in general on iOS(Chrome).

The download library used was last updated before iOS accessed the file system. 

Tried alternative libraries such as:

https://github.com/eligrey/FileSaver.js (it downloads an unnamed document without an extension)

https://github.com/kennethjiang/js-file-download (it doesn't work in Safari either)

The <a download /> tag also brought no results.
___
The bug itself is probably due to restrictions in Chrome since version 86, although, on some sites, both checked (like Gdrive) and random - files are downloaded, probably they come from the server and not formed at the front. 

Would be glad to hear any opinion on this. Maybe @Wmaileq's solution is true, but sending files back and forth is not an optimized solution

**Comment by AlexK-ino** [comment]
So far, the only result I've been able to achieve is this:
In chrome, it downloads the document without the extension
Saw an issue created on this topic, but no response to it

**Comment by jovergaag** [comment]
Hi @AlexK-ino,

Very interesting. Have you tried https://www.npmjs.com/package/js-file-downloader - it looks like it supports IOS browsers and has been updated recently.

LMK how that goes and if it doesn't work I'll check with the product team about downloading an unnamed document.

Thank you!
Jen

**Comment by AlexK-ino** [comment]
Jen, I tried the above library and this is what I got:
This solution takes URL.

I tried to form the URL using base64 and blob.

Blob: safari gives the error "WebKit blob resource error 1" and chrome identical.
(The[ file-saver](https://www.npmjs.com/package/file-saver) library uses a similar solution and gets an identical error)

Base64: iPadOS downloads the document without the extension, but if you specify the .zip extension manually, the file is working. macOS downloads with the normal file name.
(Tried naming it with URL - didn't work)

At the moment there is one questionable solution, I implemented it in the last commit of the [branch](https://github.com/Recidiviz/pulse-dashboard/tree/alexk-export-data-on-ipad) it is not fully implemented, just made for you to look at it and give permission or an alternative solution.

In short - to make a directory on the server, which will be downloaded zip, and from there to take the file to download, then deleting.

Thank you,
Alexander

**Comment by jovergaag** [comment]
Hi Alex - I think thats fine. It is definitely not ideal but since it is the only solution left it is the best one we have!

I did try to download something on your branch and got an error trying to open the zip file. I'm sure you are already aware of the problem and just haven't gotten that far yet, I just wanted to bring it to your attention!

![Screen Shot 2021-02-18 at 6 30 15 PM](https://user-images.githubusercontent.com/5872651/108448895-689f0b80-7217-11eb-8246-8bdfcc24a1e7.png)

Thank you for all of your investigation!
